src/strategies/agave/index.ts

```diff
@@ -0,0 +1,62 @@
+import { formatUnits } from '@ethersproject/units';
+import { strategy as erc20BalanceOfStrategy } from '../erc20-balance-of';
+import { call } from '../../utils';
+
+export const author = 'maxaleks';
+export const version = '0.1.0';
+
+const abi = [
+  'function balanceOf(address account) external view returns (uint256)',
+  'function totalSupply() external view returns (uint256)',
+  'function decimals() external view returns (uint256)'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const [
+    availableLiquidity,
+    lpTokenTotalSupply,
+    lpTokenDecimals,
+    underlyingTokenDecimals
+  ] = await Promise.all([
```

Comment:
You could do a multicall here instead of sending 4 requests, that would resolve faster, here is an example https://github.com/snapshot-labs/snapshot.js/blob/master/test/multicaller.ts